pkg/component/component.go

```diff
@@ -66,28 +66,38 @@ func CreateFromGit(client *occlient.Client, name string, ctype string, url strin
 	// save source path as annotation
 	annotations := map[string]string{componentSourceURLAnnotation: url}
 
-	output, err := client.NewAppS2I(name, ctype, url, labels, annotations)
+	err = client.NewAppS2I(name, ctype, url, labels, annotations)
 	if err != nil {
-		return "", errors.Wrapf(err, "unable to create git component %s", name)
+		return errors.Wrapf(err, "unable to create git component %s", name)
 	}
-	return output, nil
+
+	//get the latest build name for following
+	buildName, err := client.GetLatestBuildName(name)
+	if err != nil {
+		errors.Wrap(err, "unable to create from git")
+	}
+	err = client.FollowBuildLog(buildName)
```

Comment:
I think this should be a part of `NewAppS2I()` instead of being called after that. This should be called right after the BuildConfig has been created in `NewAppS2I()`. We can also add a `stream bool` input parameter to `NewAppS2I()` which will set if the logs should be streamed or not.
Another way could be to break `NewAppS2I()` into smaller functions, one to create DC, another for BC, which can be reused at other places.

Any way, this should be inside `NewAppS2I()`

Reply:
I don't think this should go to `NewAppS2I`.
It will just add more logic to that function also because of this difference a between builds from git and from dir.

NewAppS2I should just create objects nothing more

Reply:
It is your code @mik-dass, I'm leaving it up to you how you want to do it.